rpc/src/v1/tests/mocked/eth.rs

```diff
@@ -317,10 +317,27 @@ fn rpc_blocks_filter() {
 
 	tester.client.add_blocks(2, EachBlockWith::Nothing);
 
+	let hash1 = tester.client.block_hash(BlockId::Number(1)).unwrap();
+	let hash2 = tester.client.block_hash(BlockId::Number(2)).unwrap();
 	let response = format!(
 		r#"{{"jsonrpc":"2.0","result":["0x{:x}","0x{:x}"],"id":1}}"#,
-		tester.client.block_hash(BlockId::Number(1)).unwrap(),
-		tester.client.block_hash(BlockId::Number(2)).unwrap());
+		hash1,
+		hash2);
+
+	assert_eq!(tester.io.handle_request_sync(request_changes), Some(response.to_owned()));
+
+	// in the case of a re-org we get same block number if hash is different - BlockId::Number(2)
+	tester.client.blocks.write().remove(&hash2).unwrap();
+	tester.client.numbers.write().remove(&2).unwrap();
+	tester.client.numbers.write().insert(1, hash2);
```

Comment:
Seems that this line is actually changing hash of block1, why isn't that block part of the response?

Reply:
Good catch, that would be the case if the hash was different, but actually in ```import_block()``` in ```test_client.rs```,  that value gets overwritten again by the old hash! So I just need to take that line out because it effectively does nothing.